base/expr.jl

```diff
@@ -61,6 +61,16 @@ Takes the expression `x` and returns an equivalent expression with all macros re
 """
 macroexpand(x::ANY) = ccall(:jl_macroexpand, Any, (Any,), x)
 
+"""
+    @macroexpand
+
+Return equivalent expression with all macros removed (expanded).
```

Comment:
It should be documented that the expansion happens in the module where the `@macroexpand` macro is expanded (i.e. the same module if the expression is used directly) and not the current module when the returned code runs (which is the module if you call `macroexpand`) The difference is demonstrated below

```jl
julia> module M
       macro macroexpand(code)
           QuoteNode(macroexpand(code))
       end
       macro m()
           1
       end
       function f()
           (@macroexpand(@m), macroexpand(:(@m)))
       end
       end
M

julia> macro m()
           2
       end
@m (macro with 1 method)

julia> M.f()
(1,2)
```

This makes no difference when used in the REPL and [as I said](https://github.com/JuliaLang/julia/issues/18240#issuecomment-249354148) the current `@macroexpand` might be slightly better but this should still be documented.